browser/net/brave_static_redirect_network_delegate_helper.cc

```diff
@@ -15,6 +17,25 @@
 
 namespace brave {
 
+namespace {
+
+char safebrowsing_api_endpoint_for_testing_[] = "";
+
+std::string GetSafeBrowsingEndpoint() {
+  std::string testing_endpoint(safebrowsing_api_endpoint_for_testing_);
+  if (testing_endpoint.empty())
+    return SAFEBROWSING_ENDPOINT;
+  return testing_endpoint;
+}
+
+}  // namespace
+
+void SetSafeBrowsingEndpointForTesting(std::string endpoint) {
+  std::copy(endpoint.begin(),
```

Comment:
This is a memory overrun - there is only one byte of storage in `safebrowsing_api_endpoint_for_testing_`
I suggest to just hardcode a constant testing endpoint, and do something like `EnableTestingSafebrowsingEndpoint(true)`